### PR TITLE
Adding keyboard signum 3.1

### PR DESCRIPTION
**Pull request by immeraufdemhund**
Add keyboard layout for Signum 3.1 keyboard by Troy Fletcher
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

### CODE REVIEW

File: keyboards/signum/3_1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,14 @@
+MCU = atmega32u4
+F_CPU = 16000000
```
Comment (Rionlion100) [inline keyboards/signum/3_1/rules.mk:2]: I'm pretty sure this line isn't necessary

File: keyboards/signum/3_1/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include "kb.h"
```
Comment (Rionlion100) [inline keyboards/signum/3_1/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/signum/3_1/kb.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#pragma once
```
Comment (Rionlion100) [inline keyboards/signum/3_1/kb.h:1]: this file should be named signum.h

File: keyboards/signum/3_1/kb.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "quantum.h"
+
+// LAYOUT_KEYMAP is for there being 2 thumb cluster keys and 3 keys below alphas
+#define LAYOUT_KEYMAP( \
```
Comment (Rionlion100) [inline keyboards/signum/3_1/kb.h:6]:
```suggestion
#define LAYOUT( \
```

File: keyboards/signum/3_1/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	LAYOUT_KEYMAP(
```
Comment (Rionlion100) [inline keyboards/signum/3_1/keymaps/default/keymap.c:5]:
```suggestion
	LAYOUT(
```

File: keyboards/signum/3_1/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+		KC_MINS, KC_1   , KC_2   , KC_3   , KC_4   , KC_5   ,     KC_6   , KC_7   , KC_8   , KC_9   , KC_0   , KC_EQL ,
+		KC_TAB , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   ,     KC_Y   , KC_U   , KC_I   , KC_O   , KC_P   , KC_BSPC,
+		KC_LALT, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   ,     KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN, KC_QUOT,
+		KC_LSFT, KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   ,     KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_ENT ,
+		KC_P1  , KC_P2  , KC_P3  , KC_P4  , KC_P5  , KC_P6  ,     KC_P7  , KC_P8  , KC_P9  , KC_PDOT, KC_PEQL,KC_PENT),
+
+	LAYOUT_KEYMAP(
```
Comment (Rionlion100) [inline keyboards/signum/3_1/keymaps/default/keymap.c:12]:
```suggestion
	LAYOUT(
```

File: keyboards/signum/3_1/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+		_______, _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______,_______,
+		_______, _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______,_______,
+		_______, _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______,_______,
+		_______, _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______,_______,
+		_______, _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______,_______),
+
+	LAYOUT_KEYMAP(
```
Comment (Rionlion100) [inline keyboards/signum/3_1/keymaps/default/keymap.c:19]:
```suggestion
	LAYOUT(
```

File: keyboards/signum/3_1/kb.c [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#include "kb.h"
```
Comment (Rionlion100) [inline keyboards/signum/3_1/kb.c:1]: This file should be named signum.c

File: keyboards/signum/3_1/config.h [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    troy fletcher
+#define PRODUCT         signum 3.1
+#define DESCRIPTION     ortholinear small keyboard
```
Comment (Rionlion100) [inline keyboards/signum/3_1/config.h:11]: perhaps `small ortholinear keyboard` would make more sense here

File: keyboards/signum/3_1/config.h [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 10
+#define MATRIX_ROW_PINS { D2, D1, F5, B5, B2, C6 }
+#define MATRIX_COL_PINS { B4, D7, D0, E6, D4, F6, F4, F7, B1, B3 }
+#define DEBOUNCING_DELAY 5
+#define DIODE_DIRECTION COL2ROW
```
Comment (zvecr) [inline keyboards/signum/3_1/config.h:17]:
```suggestion
#define DIODE_DIRECTION COL2ROW
#define DEBOUNCE 5
```

File: keyboards/signum/3_1/kb.c (hunk 8 above)
Comment (zvecr) [reply keyboards/signum/3_1/kb.c:1]: File should actually be named `3_1.c`

File: keyboards/signum/3_1/kb.h (hunk 3 above)
Comment (zvecr) [reply keyboards/signum/3_1/kb.h:1]: File should actually be named `3_1.h`

File: keyboards/signum/3_1/rules.mk [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes  # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes  # Audio control and System control(+450)
+CONSOLE_ENABLE = no    # Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes      # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (zvecr) [inline keyboards/signum/3_1/rules.mk:14]: Not sure what the best option is for bootloader here, but ideally this should be solved as per the 3_0 folder tree. Short term this should be aligned to the AVR template

```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

**Comment by immeraufdemhund** [comment]
I believe I've done what was asked.. hopefully I got it right :D

File: keyboards/signum/3_1/3_1.c [hunk 12]
```diff
@@ -0,0 +1,5 @@
+#include "3_1.h"
+
+void matrix_init_kb(void) {
+    matrix_init_user();
+};
```
[the remark this answers is not in the document]
Comment (immeraufdemhund) [reply keyboards/signum/3_1/3_1.c:5]: neat, I didn't know this could be removed :D thank you.

File: keyboards/signum/3_1/rules.mk [hunk 13]
```diff
@@ -0,0 +1,30 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/signum/3_1/rules.mk:3]:
```suggestion

# Bootloader selection
```

File: keyboards/signum/3_1/rules.mk [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+# Build Options
```
Comment (fauxpark) [inline keyboards/signum/3_1/rules.mk:12]:
```suggestion

# Build Options
```

File: keyboards/signum/3_1/rules.mk [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/signum/3_1/rules.mk:30]:
```suggestion
```

**Comment by immeraufdemhund** [comment]
made requested changes to rules.mk

File: keyboards/signum/3_1/config.h [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    troy fletcher
+#define PRODUCT         signum 3.1
+#define DESCRIPTION     small ortholinear keyboard
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 10
+#define MATRIX_ROW_PINS { D2, D1, F5, B5, B2, C6 }
+#define MATRIX_COL_PINS { B4, D7, D0, E6, D4, F6, F4, F7, B1, B3 }
+#define DIODE_DIRECTION COL2ROW
+#define DEBOUNE 5
```
Comment (noroadsleft) [inline keyboards/signum/3_1/config.h:17]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/signum/3_1/3_1.c [hunk 17]
```diff
@@ -0,0 +1 @@
+#include "3_1.h"
```
Comment (tzarc) [inline keyboards/signum/3_1/3_1.c:1]: Needs a license header.

File: keyboards/signum/3_1/3_1.h [hunk 18]
```diff
@@ -0,0 +1,35 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/signum/3_1/3_1.h:1]: Needs a license header.

File: keyboards/signum/3_1/3_1.h [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "quantum.h"
+
+// LAYOUT is for there being 2 thumb cluster keys and 3 keys below alphas
+#define LAYOUT_2_thumb_3_sub_alpha( \
+	K000, K001, K002, K003, K004, K504,      K505, K005, K006, K007, K008, K009, \
+	K100, K101, K102, K103, K104, K503,      K506, K105, K106, K107, K108, K109, \
+	K200, K201, K202, K203, K204, K502,      K507, K205, K206, K207, K208, K209, \
+	K300, K301, K302, K303, K304, K501,      K508, K305, K306, K307, K308, K309, \
+	K400, K401, K402, K403, K404, K500,      K509, K405, K406, K407, K408, K409  \
```
Comment (tzarc) [inline keyboards/signum/3_1/3_1.h:11]:
```suggestion
#define LAYOUT_2_thumb_3_sub_alpha(
	K000, K001, K002, K003, K004, K504,      K505, K005, K006, K007, K008, K009,
	K100, K101, K102, K103, K104, K503,      K506, K105, K106, K107, K108, K109,
	K200, K201, K202, K203, K204, K502,      K507, K205, K206, K207, K208, K209,
	K300, K301, K302, K303, K304, K501,      K508, K305, K306, K307, K308, K309,
	K400, K401, K402, K403, K404, K500,      K509, K405, K406, K407, K408, K409
```

Don't need trailing slashes on parameters, only the content.

File: keyboards/signum/3_1/3_1.h [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+	{ K500,  K501,  K502,  K503,  K504,  K505,  K506,  K507,  K508,  K509 }  \
+}
+
+// LAYOUT2 is for there being 3 thumb cluster keys and 2 keys below alphas
+#define LAYOUT_3_thumb_2_sub_alpha( \
+	K000, K001, K002, K003, K004, K504,      K505, K005, K006, K007, K008, K009, \
+	K100, K101, K102, K103, K104, K503,      K506, K105, K106, K107, K108, K109, \
+	K200, K201, K202, K203, K204, K502,      K507, K205, K206, K207, K208, K209, \
+	K300, K301, K302, K303, K304, K501,      K508, K305, K306, K307, K308, K309, \
+	K400, K402, K403, K404, K500, K401,      K408, K509, K405, K406, K407, K409  \
```
Comment (tzarc) [inline keyboards/signum/3_1/3_1.h:27]:
```suggestion
#define LAYOUT_3_thumb_2_sub_alpha(
	K000, K001, K002, K003, K004, K504,      K505, K005, K006, K007, K008, K009,
	K100, K101, K102, K103, K104, K503,      K506, K105, K106, K107, K108, K109,
	K200, K201, K202, K203, K204, K502,      K507, K205, K206, K207, K208, K209,
	K300, K301, K302, K303, K304, K501,      K508, K305, K306, K307, K308, K309,
	K400, K402, K403, K404, K500, K401,      K408, K509, K405, K406, K407, K409
```

Don't need trailing slashes on parameters, only the content.

File: keyboards/signum/3_1/config.h [hunk 21]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/signum/3_1/config.h:1]: Needs a license header.

File: keyboards/signum/3_1/config.h [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    troy fletcher
+#define PRODUCT         signum 3.1
+#define DESCRIPTION     small ortholinear keyboard
```
Comment (tzarc) [inline keyboards/signum/3_1/config.h:11]:
```suggestion
```

File: keyboards/signum/3_1/keymaps/default/keymap.c [hunk 23]
```diff
@@ -0,0 +1,26 @@
+#include QMK_KEYBOARD_H
```
Comment (tzarc) [inline keyboards/signum/3_1/keymaps/default/keymap.c:1]: Needs a license header.

File: keyboards/signum/3_1/rules.mk [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (tzarc) [inline keyboards/signum/3_1/rules.mk:11]:
```suggestion
```

File: keyboards/signum/3_1/rules.mk [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
```
Comment (tzarc) [inline keyboards/signum/3_1/rules.mk:28]:
```suggestion
```

File: keyboards/signum/3_1/rules.mk [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (tzarc) [inline keyboards/signum/3_1/rules.mk:31]:
```suggestion
```

**Comment by flyinggecko** [comment]
Hey @immeraufdemhund , is there a chance, this will ever get ready?
Bests

**Comment by immeraufdemhund** [comment]
I forgot all about this @flyinggecko . I don't use this keyboard any more either. I'd say probably not.

**Pull request closed without merging**